redfish-core/lib/fabric_adapters.hpp

```diff
@@ -173,11 +173,36 @@ inline void
     });
 }
 
+inline void linkAsPCIeDevice(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                             const std::string& fabricAdapterPath)
+{
+    const std::string pcieDeviceName =
+        sdbusplus::message::object_path(fabricAdapterPath).filename();
+
+    if (pcieDeviceName.empty())
+    {
+        BMCWEB_LOG_ERROR ("Failed to find / in pcie device path");
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    nlohmann::json& deviceArray = asyncResp->res.jsonValue["Links"]["PCIeDevices"];
+    deviceArray = nlohmann::json::array();
+
+    deviceArray.push_back(
+        {{"@odata.id",
+          "/redfish/v1/Systems/system/PCIeDevices/" + pcieDeviceName}});
+
+    asyncResp->res.jsonValue["Links"]["PCIeDevices@odata.count"] =
+        deviceArray.size();
```

Comment:
I think L189-L194 seems better to restructure so that 
- construct the content in a tempArray
- Move it to asyncResp->res.

```
nlohmann::json::array_t deviceArray;
... <<< construct deviceArray.> .....
asyncResp->res.jsonValue["Links"]["PCIeDevices"] = std::move(deviceArray);
```

Reply:
Ack